pkg/skaffold/util/port.go

```diff
@@ -71,17 +71,17 @@ func GetAvailablePort(port int, forwardedPorts ForwardedPorts) int {
 	return p
 }
 
-func getPortIfAvailable(p int, forwardedPorts ForwardedPorts) bool {
+func getPortIfAvailable(address string, p int, forwardedPorts ForwardedPorts) bool {
 	_, loaded := forwardedPorts.LoadOrStore(p, struct{}{})
 	if loaded {
 		return false
 	}
 
-	return IsPortFree(p)
+	return IsPortFree(address, p)
 }
 
-func IsPortFree(p int) bool {
-	l, err := net.Listen("tcp", fmt.Sprintf("%s:%d", Loopback, p))
+func IsPortFree(address string, p int) bool {
+	l, err := net.Listen("tcp", fmt.Sprintf("%s:%d", address, p))
```

Comment:
if the address is not localhost, it means will be making a call to remote? I am not sure, what the user experinece would be then.
I feel, if the address is not `Loopback`, we should skip the check and return immediately.

Reply:
I think we should distinguish if the address is attached to the local host, or is a distant address.  For example, making nginx listen on a distant address will make it fail with an error '"Cannot assign requested address"